api-put-object-streaming.go

```diff
@@ -434,19 +434,29 @@ func (c *Client) putObject(ctx context.Context, bucketName, objectName string, r
 
 	var md5Base64 string
 	if opts.SendContentMd5 {
-		// Create a buffer.
-		buf := make([]byte, size)
+		// Calculate md5sum.
+		hash := c.md5Hasher()
+
+		seeker, ok := reader.(io.Seeker)
```

Comment:
This is a bit complicated to make safe (and I see the code above also makes that mistake).

But you should do a `Seek(0, io.SeekCurrent)` to get the current offset. You might be given a ReadSeeker that has been read from, meaning we should not return to absolute 0 when resetting.

Reply:
That was not the intention, the intention was simply to honor io.NewSectionReader's offset input.  Made the code more safe.